WcaOnRails/app/models/registration.rb

```diff
@@ -25,21 +27,11 @@ def pending?
     end
   end
 
-  before_save :normalize_event_ids
   private def normalize_event_ids
     self.eventIds = events.map(&:id).join(" ")
   end
 
-  attr_writer :birthday
-  def birthday
-    birthYear == 0 || birthMonth == 0 || birthDay == 0 ? nil : Date.parse("%04i-%02i-%02i" % [ birthYear, birthMonth, birthDay ])
-  end
-
-  before_validation :unpack_dates
   private def unpack_dates
-    if @birthday.nil? && !birthday.blank?
```

Comment:
Ah, crap. I read it as `@birthday.nil? && !@birthday.blank?` which always evaluates to false. That's why I removed it. Nice catch! I'll revert the change.